devito/finite_differences/differentiable.py

```diff
@@ -234,18 +235,19 @@ def __new__(cls, obj):
 
     def _doit(obj, args=None):
         cls = diffify._cls(obj)
+        args = args or obj.args
+
         if cls is obj.__class__:
+            # Try to just update the args if possible (Add, Mul)
+            if getattr(obj, 'is_Add', False):
```

Comment:
the try-except was way neater

avoid `getattr` at all costs when possible

Reply:
This is only for Add and the try/except would catch mul wich creates problems

Reply:
Two things

1. that's not what the comment above says (Add, Mul)

2. if it were just for Add, it'd smell like a hack

Reply:
Ok so it doesn't break with both, but the dse doesn't like the expression anymore and end up with like double opcount because of it in the tti tests. So it is a hack but not because of sympy but because factorization isn't detected if I put back mul in it

Reply:
So I can put back the try/except which is the one that should be done but I'll have to change the opcounts in way higher values in test_dse (the 1/h_x are not factorized anymore for example)

Reply:
we should fix the factorizer